head.go

```diff
@@ -969,6 +969,23 @@ func (h *headIndexReader) LabelValues(names ...string) (index.StringTuples, erro
 	return index.NewStringTuples(sl, len(names))
 }
 
+// LabelNames returns all the unique label names present in the head.
+func (h *headIndexReader) LabelNames() ([]string, error) {
+	h.head.symMtx.RLock()
+	defer h.head.symMtx.RUnlock()
+	labelNames := make([]string, 0, len(h.head.values))
+	for name := range h.head.values {
+		if name == "" {
+			continue
+		}
+		labelNames = append(labelNames, name)
+	}
+	sort.Slice(labelNames, func(i, j int) bool {
+		return labelNames[i] < labelNames[j]
+	})
+	return labelNames, nil
```

Comment:
you can use directly the strings sort func.

```suggestion
	return sort.Strings(labelNames), nil
```